src/core/on-uncaught-exception.js

```diff
@@ -0,0 +1,52 @@
+import config from "./config";
+import ProcessingQueue from "./processing-queue";
+import { sourceFromStacktrace } from "./stacktrace";
+import { extend } from "./utilities";
+import Logger from "../logger";
+import { test } from "../test";
+
+/**
+ * Handle a global error that should result in a failed test run.
+ *
+ * Summary:
+ *
+ * - If there is a current test, it becomes a failed assertion.
+ * - If there is a current module, it becomes a failed test (and bypassing filters).
+ *   Note that if we're before any other test or module, it will naturally
+ *   become a global test.
+ * - If the overall test run has ended, the error is printed to `console.warn()`.
+ *
+ * @since 2.17.0
+ * @param {Error|any} error
+ */
+export default function onUncaughtException( error ) {
+	const message = ( error.message ? error.toString() : error );
```

Comment:
Don't we have a pretty-print error messaging utility somewhere? Worth reusing? I remember there were some legacy use cases to retain there

Reply:
Changed to re-use `errorString()` as used by `assert.throws()` for turning any kind of thrown value into a string.